OpenSim/Framework/Console/ConsoleMode.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace OpenSim.Framework.Console
+{
+    public static class ConsoleMode
+    {
+        const uint ENABLE_QUICK_EDIT = 0x0040;
+        const uint ENABLE_EXTENDED_FLAGS = 0x0080;
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern IntPtr GetStdHandle(int nStdHandle);
+
+        [DllImport("kernel32.dll")]
+        static extern bool GetConsoleMode(IntPtr hConsoleHandle, out uint lpMode);
+
+        [DllImport("kernel32.dll")]
+        static extern bool SetConsoleMode(IntPtr hConsoleHandle, uint dwMode);
+
+        private const int STDIN_HANDLE = -10;
+        internal static bool DisableQuickEdit()
+        {
+            IntPtr consoleHandle = GetStdHandle(STDIN_HANDLE);
+
+            uint consoleMode;
+            // get current console mode
+            if (!GetConsoleMode(consoleHandle, out consoleMode))
+                return false;
+
+            // setting QUICK_EDIT mode either way requires EXTENDED
+            consoleMode |= ENABLE_EXTENDED_FLAGS;
+            consoleMode &= ~ENABLE_QUICK_EDIT;
+            return SetConsoleMode(consoleHandle, consoleMode);
+        }
+    }
+}
```

Comment:
Looking over the history I found that similar logic was removed in 4bd35516689a8e2591eed964fd9f48fceb410091. However that version set more flags than this does. Maybe useful?

Reply:
The extra flags were a bit overzealous from a time when Halcyon development was almost exclusively for InWorldz' us (at least the insert mode one is). Setting the one flag resolves the issue, so I'd like to keep it focused on solving one problem and not tying that to other changes that may not be necessary.  If this turns out to be an incomplete solution for 100% of the problem cases, we can take a look at also adding back the disabling of MOUSE_INPUT and WINDOW_INPUT (at that time).